Source/WebCore/css/calc/CSSCalcValue.cpp

```diff
@@ -188,7 +189,7 @@ inline double CSSCalcValue::clampToPermittedRange(double value) const
     if (m_tree.category == Calculation::Category::Integer)
         value = std::floor(value + 0.5);
 
-    return m_tree.range == ValueRange::NonNegative && value < 0 ? 0 : value;
```

Comment:
I normally avoid `std::clamp` because it returns `max` if `max` < `min`, and we typically want `min` in those cases in most call sites. But if that is not an issue here because `CSS::Range` never violates the invariant, then I think it's great to use it.